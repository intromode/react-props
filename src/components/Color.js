import React from 'react';
import PropTypes from 'prop-types';



export default function Color({ name, hex, rgb }) {
  const divStyle = {
    backgroundColor: hex
  };

  return (
    <dl style={divStyle}>
      <dt>Name</dt>
      <dd>{name ? name : hex}</dd>

      <dt>Hex</dt>
      <dd>{hex}</dd>

      <dt>rgb</dt>
      <dd>{rgb}</dd>
    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string,
  hex: PropTypes.string.isRequired,
  rgb: PropTypes.string.isRequired,
};
