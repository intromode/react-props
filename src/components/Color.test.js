import React from 'react';
import Color from './Color';
import { shallow } from 'enzyme';

describe('testing the color component', () => {
  it('tests the snapshot', () => {
    const wrapper = shallow(<Color name="purple" hex="#9400D3" rgb="fkfjkdf" />);
    expect(wrapper).toMatchSnapshot();
  });
});
