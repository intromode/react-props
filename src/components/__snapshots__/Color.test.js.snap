// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the color component tests the snapshot 1`] = `
<dl
  style={
    Object {
      "backgroundColor": "#9400D3",
    }
  }
>
  <dt>
    Name
  </dt>
  <dd>
    purple
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    #9400D3
  </dd>
  <dt>
    rgb
  </dt>
  <dd>
    fkfjkdf
  </dd>
</dl>
`;
